backend/nodes/properties/inputs/numeric_inputs.py

```diff
@@ -0,0 +1,144 @@
+from .base_input import BaseInput
+
+
+class NumberInput(BaseInput):
+    """Input a number"""
+
+    def __init__(
+        self,
+        label: str,
+        default=0.0,
+        minimum=0,
+        maximum=None,
+        step=1,
+        optional=False,
+        number_type="any",
+    ):
+        super().__init__(f"number::{number_type}", label)
+        self.default = default
+        self.minimum = minimum
+        self.maximum = maximum
+        self.step = step
+        self.optional = optional
+
+    def toDict(self):
+        return {
+            "type": self.input_type,
+            "label": self.label,
+            "min": self.minimum,
+            "max": self.maximum,
+            "def": self.default,
+            "step": self.step,
+            "hasHandle": True,
+            "optional": self.optional,
+        }
+
+    def enforce(self, value):
+        return max(float(self.minimum), float(value))
+
+
+class IntegerInput(NumberInput):
+    """Input an integer number"""
+
+    def __init__(self, label: str):
+        super().__init__(label, default=0, minimum=0, maximum=None, step=None)
+
+    def enforce(self, value):
+        return max(int(self.minimum), int(value))
+
+
+class BoundedNumberInput(NumberInput):
+    """Input for a bounded float number range"""
+
+    def __init__(
+        self,
+        label: str,
+        minimum: float = 0.0,
+        maximum: float = 1.0,
+        default: float = 0.5,
+        step: float = 0.25,
+    ):
+        super().__init__(
+            label, default=default, minimum=minimum, maximum=maximum, step=step
+        )
+
+    def enforce(self, value):
+        return min(max(float(self.minimum), float(value)), float(self.maximum))
+
+
+class OddIntegerInput(NumberInput):
+    """Input for an odd integer number"""
+
+    def __init__(self, label: str, default: int = 1, minimum: int = 1):
+        super().__init__(label, default=default, minimum=minimum, maximum=None, step=2)
+
+    def enforce(self, value):
+        odd = int(value) - (int(value) % 2)
```

Comment:
That's interesting. So you round *down* to the nearest odd number. The front end makes a similar correction, but it rounds *up* to the nearest odd number.

Reply:
Let's use the front end version.

I also just notice that this implementation doesn't actually work.
```py
>>> def make_odd(n):
...     return n - (n % 2)
...
>>> make_odd(3)
2
>>> make_odd(4)
4
>>> make_odd(5)
4
>>> make_odd(6)
6
```

Reply:
Oh oops, that should be `n - (1 - (n% 2))`